File: server/MovieSocialNetwork/DomainCore/src/main/kotlin/models/FavoriteItemList.kt
package models

import org.jetbrains.annotations.NotNull
import java.sql.Timestamp
import java.util.UUID

data class FavoriteItemList(
  val listId: UUID = UUID.randomUUID(),
  @NotNull val userId: UUID,
  @NotNull val name: String,
  @NotNull val type: ItemType,
  var items: MutableSet<Item> = HashSet(),
  @NotNull val timestamp: Timestamp,
)

File: client/src/app/model/domain/favorite-item-list.ts
import { Item } from "./item"

export interface FavoriteItemList {
  listId?: string,
  userId?: string,
  name?: string,
  type?: ListType,
  items: Item[],
  timestamp: Date
}

export enum ListType {
  MOVIE= 'MOVIE',
  STAR = 'STAR',
  TV_SERIES = 'TV_SERIES'
}
